docs/coffeescript.js

```diff
@@ -0,0 +1,309 @@
+const coffeeSelector = document.getElementById("coffee-Selector");
+const teaSelector = document.getElementById("tea-Selector");
+const dessertSelector = document.getElementById("dessert-Selector");
+const grid = document.getElementById("menu-grid");
+const menuGrid = document.getElementById("menu-grid");
+const menuButton = document.getElementById("coffee-menu");
+const coffee_modal = document.getElementById(".modal");
+const navModal = document.querySelector(".nav-modal");
+let total;
+let lastValue;
+let products;
+let productsLength;
+
+// reading the material.JSON file
+async function fetchData() {
+  const response = await fetch("./material.json");
+  products = await response.json();
+  productsLength = products.length;
+  createMenuGrid(coffeeSelector.value);
+}
+
+fetchData();
+
+// script for creating grids
+function createMenuGrid(x) {
+  let gridContent = "";
+  let imageIndex = 1;
+  for (let i = 0; i < productsLength; i++) {
+    if (products[i].category === x) {
+      gridContent += ` 
+                    <div class="preview" id="previewCard${
+                      i + 1
+                    }" onclick="openModal('${x}',${imageIndex},${i})">
+                    <img
+                        src="./images/${x}-${imageIndex}.jpg"
+                        alt="Photo of a ${x}"
+                        class="preview image" />
+                    <h3 class="price-text coffee-name">${products[i].name}</h3>
+                    <p class="coffee-description">
+                    ${products[i].description}
+                    </p>
+                    <h3 class="price-text">$${products[i].price}</h3>
+                    </div>`;
+      imageIndex++;
+    }
+  }
+  return (grid.innerHTML = gridContent);
+}
+
+coffeeSelector.addEventListener("click", (e) => {
+  e.preventDefault();
+  createMenuGrid(coffeeSelector.value);
+});
+
+teaSelector.addEventListener("click", (e) => {
+  e.preventDefault();
+  createMenuGrid(teaSelector.value);
+});
+
+dessertSelector.addEventListener("click", (e) => {
+  e.preventDefault();
+  createMenuGrid(dessertSelector.value);
+});
+
+createMenuGrid(coffeeSelector.value);
```

Comment:
Is this still necessary? you do it in fetchData function

Reply:
Line 65 is unnecessary. Therefore it is removed now.